coordinator/src/trade/mod.rs

```diff
@@ -372,24 +362,65 @@ impl TradeExecutor {
         let event_id = format!("{contract_symbol}{maturity_time}");
 
-        let (offer_collateral, accept_collateral, fee_config) = match trader_required_utxos {
-            TraderRequiredLiquidity::ForTradeCostAndTxFees => (
-                margin_coordinator + collateral_reserve_coordinator.to_sat(),
-                margin_trader + collateral_reserve_trader + order_matching_fee,
-                dlc::FeeConfig::EvenSplit,
-            ),
-            TraderRequiredLiquidity::None => (
-                margin_coordinator
-                    + collateral_reserve_coordinator.to_sat()
-                    + margin_trader
-                    + collateral_reserve_trader
-                    // If the trader doesn't bring their own UTXOs, including the order matching fee
-                    // is not strictly necessary, but it's simpler to do so.
-                    + order_matching_fee,
-                0,
-                dlc::FeeConfig::AllOffer,
-            ),
-        };
+        let (offer_collateral, accept_collateral, fee_config, collateral_reserve_trader) =
+            match trader_required_utxos {
+                TraderRequiredLiquidity::ForTradeCostAndTxFees => (
+                    margin_coordinator + collateral_reserve_coordinator.to_sat(),
+                    margin_trader + collateral_reserve_trader + order_matching_fee,
+                    dlc::FeeConfig::EvenSplit,
+                    collateral_reserve_trader,
+                ),
+                TraderRequiredLiquidity::None {
+                    external_funding_sat,
+                } => {
+                    // if the user funded the channel externally we derive the collateral reserve
+                    // trader from the difference of the trader margin and the
+                    // externally received funds.
+                    let collateral_reserve_trader = match (margin_trader + order_matching_fee)
+                        .cmp(&external_funding_sat)
+                    {
+                        // FIXME(holzeis): At the moment we do not consider the funding tx fee nor
+                        // the channel reserve fee here. The user funds it, but we only use it as
+                        // buffer.
+                        Ordering::Less => external_funding_sat - margin_trader - order_matching_fee,
+                        Ordering::Equal => 0,
+                        // TODO(holzeis): Introduce margin orders to directly use the
+                        // external_funding_sats for the position instead of failing here. We need
+                        // to do this though as a malicious actor could otherwise drain us.
+                        //
+                        // Note, we add a min trader reserve to the external funding to ensure that
+                        // minor price movements are covered.
+                        Ordering::Greater => bail!("Not enough external funds to open position."),
+                    };
```

Comment:
🔧 It would be simpler to use 

```rust
external_funding_sat.checked_sub(margin_trader + order_matching).context("Not enough external funds to open position")?
```